admin/services/pin-generation.service.js

```diff
@@ -90,30 +94,39 @@ pinGenerationService.generatePupilPins = async (pupilsList) => {
 pinGenerationService.generateSchoolPassword = (school) => {
   let { schoolPin, pinExpiresAt } = school
   if (!pinValidator.isActivePin(schoolPin, pinExpiresAt)) {
-    const length = 8
-    school.schoolPin = pinGenerationService.generateRandomPin(length)
+    const allowedWords = (config.Data.allowedWords && config.Data.allowedWords.split(',')) || []
+    const firstRandomWord = allowedWords.length > 0
```

Comment:
These ternary expressions are duplicated.  It make more sense, IMHO, to palace a guard condition at the start of the function to ensure that allowedWords is populated.